### DISCUSSION TITLE
How to get the values of a parameter in Arbor?

### DISCUSSION THREAD

**Discussion by lungsi** [arbor-sim/arbor#1209] [Q&A]
`Arbor Version: 0.4.1.dev0`

# Getting parameter values for global properties (of a cell)
Let us assume we have set the properties of a cell with
```
cell.set_properties(Vm=-70, cm=1.0, rL=500., tempK=30+273.5)
cell.set_ion(ion="na", rev_pot=50.)
cell.set_ion(ion="k", rev_pot=-77.)
#
leaky_chnl = arbor.mechanism('pas', {'e': -54.3, 'g': 0.0003})
cell.paint('"everywhere"', leaky_chnl)
```

## How can I get (or view) the parameter values of say, `Vm`, `cm`, `rL`, `tempK`, `rev_pot` for respective ions, reversal potential and conductance of the passive mechanism?

# Getting parameter values for regional properties (of a cell)
Again, assume the properties of regions of a cell set as
```
hh_na_k_chnl = arbor.mechanism("hh", {"gnabar": 0.120,
                                      "gkbar": 0.036,
                                      "gl": 0.})
cell.paint('(join (region "soma") (region "dend") (region "axon"))', "na", rev_pot=50)
cell.paint('(join (region "soma") (region "dend") (region "axon"))', "k", rev_pot=-77)
cell.paint('(join (region "soma") (region "dend") (region "axon"))', hh_na_k_chnl)
```

## How do I get the parameter values of the mechanisms painted in particular cell regions?

PS: The example here are for parameters set by the user but the questions also applies to accessing default parameter values as well.

**Comment by bcumming** [comment]
Hello @lungsi.

Good question, and unfortunately we don't currently provide an interface for this.
The good news is that we are currently working on a feature for viewing, modifying, saving and loading descriptions of all of the painted, placed and defaulted values/mechanisms/settings/ions/etc. on a cell. This is scheduled for inclusion in v0.5, though it will land in master before then.

I will post an update here with a link to the relevant documentation on how to do this when it is merged.

**Comment by Helveg** [comment]
@bcumming any updates?

**Comment by thorstenhater** [comment]
<img width="715" alt="Screenshot 2022-08-05 at 13 55 37" src="https://user-images.githubusercontent.com/24411438/183072441-1efd1c5e-79b6-4cad-9d26-1ebcc20ac6f5.png">

This works for the global properties.
